server/src/service_manager.rs

```diff
@@ -5,20 +5,23 @@ use std::{collections::HashMap, sync::Arc};
 use tokio::sync::RwLock;
 use zbus::zvariant::{ObjectPath, OwnedObjectPath};
 
-use crate::session::Session;
+use crate::{collection::Collection, session::Session};
 
 #[derive(Debug)]
 pub struct ServiceManager {
     connection: zbus::Connection,
     // sessions mapped to their corresponding object path on the bus
     sessions: HashMap<OwnedObjectPath, Arc<Session>>,
+    // collections mapped to their corresponding object path on the bus
```

Comment:
I am not a fan of duplicating this information between Service and ServiceManager. Service already knows about the collections and I don't think it makes sense to move them from there to ServiceManager. 

I see that we are pushing everything into ServiceManager because it is easy but that leads to spaghetti code. Instead, I think another approach would be to give a back reference to the Service in the ServiceManager then you would let the Service handle both the Collections and the sessions as it is the object responsible for them.

Reply:
Sorry for the delay. Implemented through: https://github.com/bilelmoussaoui/oo7/pull/165